addons/addon-base-raas/packages/base-raas-rest-api/lib/controllers/study-controller.js

```diff
@@ -157,23 +140,17 @@ async function configure(context) {
   router.put(
     '/:id/permissions',
     wrap(async (req, res) => {
+      const [studyService, lockService, environmentMountService] = await context.service([
+        'studyService',
+        'lockService',
+        'environmentMountService',
+      ]);
       const studyId = req.params.id;
       const requestContext = res.locals.requestContext;
       const updateRequest = req.body;
 
-      // Validate permissions and usage
-      // For future - Move authZ logic inside service (or new service) & move away from verifyRequestorAccess
-      await studyPermissionService.verifyRequestorAccess(requestContext, studyId, req.method);
-      const study = await studyService.mustFind(requestContext, studyId);
-      if (study.category === 'My Studies') {
-        throw context.boom.forbidden('Permissions cannot be set for studies in the "My Studies" category', true);
-      }
-      if (study.category === 'Open Data') {
-        throw context.boom.forbidden('Permissions cannot be set for studies in the "Open Data" category', true);
-      }
-
       const result = await lockService.tryWriteLockAndRun({ id: `${studyId}-workspaces` }, async () => {
-        const updateOutcome = await studyPermissionService.update(requestContext, studyId, updateRequest);
+        const updateOutcome = await studyService.updatePermissions(requestContext, studyId, updateRequest);
```

Comment:
Sanity check - Given the changes to the getStudyAccessInfo() method, does permission propagation still work as expected for upgrade scenarios?

Reply:
I didn't do an end to end test for the permission propagation.  Most of the existing logic will need to be refactored now that we have BYOB in the picture.

Reply:
I'm guessing that's a problem, since we'd be potentially breaking R/W functionality for users wishing to switch permissions back and forth on existing environments.